tfx_addons/firebase_publisher/README.md

```diff
@@ -21,25 +21,60 @@ This project helps users to publish trained models directly from TFX Pusher comp
+Firebase ML Publisher component will be implemented as Python class-based component. You can find the [actual source code](https://github.com/deep-diver/complete-mlops-system-workflow/tree/feat/firebase-publisher/training_pipeline/pipeline/components/pusher/FirebasePublisher) in my personal project. 
 
 The implementation details
-- Define a custom Python function-based TFX component. It takes the following parameters from a previous component.
-  - The URI of the pushed model from TFX Pusher component.
-  - Requirements from Firebase ML (credential JSON file path, Firebase temporary-use GCS bucket). Please find more information from [Before you begin section](https://firebase.google.com/docs/ml/manage-hosted-models#before_you_begin) in the official Firebase document.
-  - Meta information to manage published model for Firebase ML such as `display name` and `tags`.
-- Download the Firebase credential file and pushed TFLite model file.
-- Initialize Firebase Admin with the credential and Firebase temporary-use GCS bucket.
-- Search if any models with the same `display name` has already been published.
-  - if yes, update the existing Firebase ML mode, then publish it
-  - if no, create a new Firebase ML model, then publish it
-- Return `tfx.dsl.components.OutputDict(result=str)` to indicate if the job went successful, and if the job was about creating a new Firebase ML model or updating the exisitng Firebase ML model.
+- This component behaves similar to [Pusher](https://www.tensorflow.org/tfx/api_docs/python/tfx/v1/components/Pusher) component, but it pushes/hosts model to Firebase ML instead. To this end, FirebasePublisher interits Pusher, and it gets the following inputs
+
+```python
+FirebasePublisher(
+  model: types.BaseChannel = None,
+  model_blessing: Optional[types.BaseChannel] = None,
+  custom_config: Optional[Dict[str, Any]] = None,
+)
+```
+
+- Each inputs:
+  - `model` : the model from the upstream TFX component such as [Trainer](https://www.tensorflow.org/tfx/api_docs/python/tfx/v1/components/Trainer)
+  - `model_blessing` : the output of `blessing` from the Evaluator component to indicate if the given `model` is good enough to be pushed
+  - `custom_config` : additional information to initialize and configure [Firebase Admin SDK](https://firebase.google.com/docs/reference/admin/python). `FIREBASE_ML_MODEL_NAME` and `FIREBASE_ML_MODEL_TAGS` correspond to the `display_name` and `tags` respectively of the Firebase hosted model. `FIREBASE_CREDENTIALS` is a optional parameter, and it indicates GCS location where a Service Account Key (JSON) file is stored. If this parameter is not given, [Application Default Credentials](https://cloud.google.com/docs/authentication/production) will be used in GCP environment
+
+    ```python
+    FIREBASE_ML_ARGS = {
```

Comment:
nit: you can probs isntead of use a dictionary set those as arguments for the component. Its easier to do:

```
FirebasePublisher(model_name="something", model_tags=["aa","bbb"])
```
than 
```
FirebasePublisher(config={"FIREBASE_ML_MODEL_NAME":"something", ...})
```

Reply:
I like your suggestion, but it looks like any Cloud specific configurations are passed in `custom_config` for all TFX components such as `Trainer`, `Transform`, and `Pusher`. I proposed this way to be consistent with the other Components.

Reply:
Yeah, bc those components used to be able to be used for both local mode or remote mode via executor injection.

Now that executor injection is not a thing, (it was deprecated on TFX 1.0) it makes more sense to use regular python arguments to parametrize components. I don't think FirebasePublisher will ever be used outside of  pushing to Firebase, so it will make it simpler to just parametrize it in arguments.

That said, this is a nitpick, so you can ignore this

Reply:
Nice suggestion! I didn't know executor injection is deprecated.